### PR TITLE
Combine EngineError and StratisError

### PR DESCRIPTION
**Pull request by agrover**
StratisError is currently EngineError plus dbus errors. In order to support propertychanged signals (#201) from the engine, we will be needing to add dbus errors to EngineError.

Since there's no longer a difference, move the essence of EngineError into StratisError, and then use StratisError (and StratisResult) throughout all the code.

Finally, simplify declarations since these very commonly used types are now exported at the top level.

### CODE REVIEW

**Comment by mulkieran** [comment]
I have some reservations about this change which I will endeavor to expand on tomorrow.

**Comment by agrover** [comment]
rebased and conflicts fixed.

File: src/engine/strat_engine/cleanup.rs [hunk 1]
```diff
@@ -4,14 +4,15 @@
 
 // Code to handle cleanup after a failed operation.
 
-use super::super::errors::{EngineError, EngineResult, ErrorEnum};
+use stratis::{StratisError, StratisResult, ErrorEnum};
```
Comment (mulkieran) [inline src/engine/strat_engine/cleanup.rs]: alphabetize!

File: src/stratis/mod.rs [hunk 2]
```diff
@@ -3,8 +3,8 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 pub use self::stratis::VERSION;
-pub use self::errors::{StratisError, StratisResult};
+pub use self::errors::{StratisError, StratisResult, ErrorEnum};
```
Comment (mulkieran) [inline src/stratis/mod.rs]: alph!

File: src/stratis/mod.rs [hunk 3]
```diff
@@ -3,8 +3,8 @@
...
 
-mod errors;
+pub mod errors;
```
Comment (mulkieran) [inline src/stratis/mod.rs]: Remove ```pub``` modifier!

**Comment by agrover** [comment]
CI failing due only to "line too long" rustfmt error. Hopefully #876 will fix this at some point.

**Comment by mulkieran** [comment]
Just to be clear, I expected that you would fix the format error, and believed that it wasn't worth mentioning, which was why I didn't mention it.

**Pull request merged**